File: app/src/main/java/com/example/unsplash/model/datasource/CollectionDataSourceFactory.java
package com.example.unsplash.model.datasource;

import android.arch.lifecycle.MutableLiveData;
import android.arch.paging.DataSource;
import android.arch.paging.PageKeyedDataSource;

import com.example.unsplash.model.models.Collection;

public class CollectionDataSourceFactory extends DataSource.Factory {
    private MutableLiveData<PageKeyedDataSource<Integer, Collection>> collectionLiveDataSource = new MutableLiveData<>();

    @Override
    public DataSource create() {
        CollectionDataSource collectionDataSource = new CollectionDataSource();
        collectionLiveDataSource.postValue(collectionDataSource);
        return collectionDataSource;
    }

    public MutableLiveData<PageKeyedDataSource<Integer, Collection>> getCollectionLiveDataSource() {
        return collectionLiveDataSource;
    }
}

File: app/src/main/java/com/example/unsplash/model/datasource/CollectionDataSourceFactory.kt
package com.example.unsplash.model.datasource

import androidx.lifecycle.MutableLiveData
import androidx.paging.DataSource
import androidx.paging.PageKeyedDataSource
import com.example.unsplash.model.models.ColletionPhotos

class CollectionDataSourceFactory : DataSource.Factory<Int, ColletionPhotos>() {
    private val collectionLiveDataSource = MutableLiveData<PageKeyedDataSource<Int, ColletionPhotos>>()

    override fun create(): DataSource<Int, ColletionPhotos> {
        val collectionDataSource = CollectionDataSource()
        collectionLiveDataSource.postValue(collectionDataSource)
        return collectionDataSource
    }
}
